media-proxy/src/mtl.c

```diff
@@ -2474,9 +2479,43 @@ tx_st22p_session_context_t* mtl_st22p_tx_session_create(mtl_handle dev_handle, s
         st_pthread_cond_destroy(&tx_ctx->st22p_wake_cond);
         return NULL;
     }
+
     tx_ctx->handle = tx_handle;
     tx_ctx->frame_size = st22p_tx_frame_size(tx_handle);
 
+#if defined(ZERO_COPY) || defined(TX_ZERO_COPY)
+    uint8_t planes = st_frame_fmt_planes(ops_tx.input_fmt);
+    size_t frame_size = tx_ctx->frame_size;
+
+    tx_ctx->p_ext_frames = (struct st_ext_frame*)malloc(
+        sizeof(*tx_ctx->p_ext_frames) * tx_ctx->fb_cnt);
+    size_t pg_sz = mtl_page_size(dev_handle);
+    size_t fb_size = tx_ctx->frame_size * tx_ctx->fb_cnt;
+    tx_ctx->ext_fb_iova_map_sz = mtl_size_page_align(fb_size, pg_sz); /* align */
+    size_t fb_size_malloc = tx_ctx->ext_fb_iova_map_sz + pg_sz;
+    tx_ctx->ext_fb_malloc = calloc(1, fb_size_malloc);
```

Comment:
tx_ctx->ext_fb_malloc should be freeed in mtl_st22p_tx_session_destroy()?

Reply:
that's right. Fixed.